src/rules/objectTypeEolRule.ts

```diff
@@ -0,0 +1,71 @@
+import * as ts from "typescript";
+import * as Lint from "tslint/lib/lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_SEMICOLON = "object type members should end with a semicolon";
+    public static FAILURE_COMMA = "object type members should end with a comma";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new ObjectTypeWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class ObjectTypeWalker extends Lint.RuleWalker {
+    protected visitTypeLiteral(node: ts.TypeLiteralNode) {
+        node.members.forEach(member => {
+            this.checkSemicolonAt(member);
+        });
+
+        super.visitTypeLiteral(node);
+    }
+
+    protected visitInterfaceDeclaration(node: ts.InterfaceDeclaration) {
+        node.members.forEach(member => {
+            this.checkSemicolonAt(member);
+        });
+
+        super.visitInterfaceDeclaration(node);
+    }
+
+    private checkSemicolonAt(node: ts.Node) {
```

Comment:
`checkSemicolonAt` is the same name in the Semicolon rule. The functions are split because the `checkLastToken` code is reused between semicolons and commas, depending on the configured options.